src/libraries/Common/tests/System/TimeProviderTests.cs

```diff
@@ -316,12 +316,26 @@ public static void RunDelayTests(TimeProvider provider, ITestTaskFactory taskFac
             Assert.True(task1.Status == TaskStatus.RanToCompletion, "    > FAILED.  Expected Delay(TimeSpan(0), timeProvider) to run to completion");
             Assert.True(task2.Status == TaskStatus.RanToCompletion, "    > FAILED.  Expected Delay(TimeSpan(0), timeProvider, uncanceledToken) to run to completion");
 
+            // This should complete quickly with a CANCELED status.
+            Task task3 = taskFactory.Delay(provider, new TimeSpan(0), new CancellationToken(true));
+            try
+            {
+                var e = Assert.Throws<AggregateException>(task3.Wait);
+                Assert.IsAssignableFrom<OperationCanceledException>(e.InnerException);
+            }
+            catch (Exception e)
```

Comment:
do we need the try-catch at all? shouldn't `var e = Assert.Throws<AggregateException>(task3.Wait);` will validate the thrown exception and will fail if got different exception? did you try that?

Reply:
This was also just to remain consistent with the rest of the test, which is converting thrown exceptions into an `Assert.Fail` with a specific message. I presume this is because a single test method is testing multiple cases and it helps to identify which case failed in logs. However, you are correct, with the exception of more detailed logging, the Asserts alone are sufficient and the `try..catch` is not required. Happy to implement this whatever way you feel is best.

Reply:
The other try-catch in the file is not using `Assert.Throws` inside the try block. If you prefer to assert with a good message, you may consider using `Record.Exception` instead. here is example of that https://github.com/dotnet/runtime/blob/959c10e41f1180482a7bc25e2bba16d27aca343a/src/libraries/System.Data.Odbc/tests/ReaderTests.cs#L28.